contributors/design-proposals/admission_control_event_rate_limit.md

```diff
@@ -0,0 +1,155 @@
+# Admission control plugin: EventRateLimit
+
+## Background
+
+This document proposes a system for using admission control to enforce a limit
+on the number of event requests that the API Server will accept in a given time
+slice. In a large multi-tenant cluster, there may be a small percentage of
+tenants that are always in some type of error state, in which each tenant is
+producing a steady stream of error event requests. Each individual tenant may not
+be producing an absurd amount of event requests on its own, but taken collectively
+the errors from this small percentage of tenants can have a significant impact on
+the performance of the cluster overall. 
+
+## Use cases
+
+1. Ability to protect the API Server from being flooded by event requests.
+2. Ability to protect the API Server from being flooded by event requests for
+   a particular namespace.
+3. Ability to protect the API Server from being flooded by event requests for
+   a particular user.
+4. Ability to protect the API Server from being flooded by event requests from
+   a particular source+object.
+
+## Data Model
+
+### Configuration
+
+```go
+// Type of limit (e.g., per-namespace)
+type LimitType string
+
+const (
+	// ServerLimitType limits are maintained against all events received by the server
+	ServerLimitType LimitType = "server"
+	// NamespaceLimitType limits are maintained against events from each namespace
+	NamespaceLimitType LimitType = "namespace"
+	// UserLimitType limits are maintained against events from each user
+	UserLimitType LimitType = "user"
+	// SourceObjectLimitType limits are maintained against events from each source+object
+	SourceObjectLimitType LimitType = "source+object"
+)
+
+// Configuration provides configuration for the EventRateLimit admission controller.
+type Configuration struct {
```

Comment:
More specific type names, such as EventRateLimiter, would be useful.

Reply:
The other API Server admission controllers with configuration--podtolerationrestriction and resroucequota--both use the name Configuration, so I assumed that was the preferred name for the configuration type. If that is not the case, then I am more than happy to change to a more specific name.